### ISSUE TITLE
Linux

### ISSUE THREAD

**Issue by lightbulblighter** [lightbulblighter/Rboxlo#56]
- CMake
- Bash api-key generator
- .NET Core

**Comment by lightbulblighter** [comment]
As it stands, here's where Linux support is now:

- dotNet doesn't seem to like being compiled on Unix systems. Boo. [I saw this, though.](https://stackoverflow.com/questions/41533592/how-to-compile-net-core-app-for-linux-on-a-windows-machine) Maybe it'll teach me something. I don't think we'll necessarily *need* CMake, but it certainly is a possibility.
- I have pruned most of the code that used libraries such as `System.Windows.Forms`, which uses Win32 directly. I also decided to re-implement some archaic C# system libraries to be better. They probably have Windows-dependent code lurking in them, so I'd rather not take risks. However, at the same time, parts of the Launcher now directly access User32 to do some shenanigans such as removing the close button from a dialog (WOW!). This results in no net change. Oooops.
- I very much doubt that we will be able to go to .NET Core in the release of v1.0.0. I've modernized a lot of the code to use new fancy C# libraries (aka libraries that weren't implemented decades ago) such as swapping WebClients for WebRequests... which I should've done in the first place. .NET 5.0 seems extremely promising as it says to have .NET Core support for WPF. We might be able to actually get Linux users. I love you, Torvalds.
- The Arbiter, which is supposed to be RccServiceArbiter's brain-damaged little brother, relies on many Windows technologies. After "careful consideration" I just decided to rewrite the entire thing and make it Unix dependent. Probably might throw in WINE launching for gameservers. Would be awesome to run the Arbiter on Linux.

The gist of it is that I've taken two steps forward to implement Linux and probably one and a half steps backward. The most possible Linux support for v1.0.0 is for the Arbiter. The Website already has Linux support thankfully so I needn't do much more to it.
